content/en/dynamic_instrumentation/_index.md

```diff
@@ -23,110 +23,166 @@ further_reading:
+- [**Gauge**][2]: Generates a gauge based on the last value of a variable. This metric requires a [metric expression](#expression-language).
+- [**Histogram**][3]: Generates a statistical distribution of a variable. This metric requires a [metric expression](#expression-language).
 
 To create a metric probe:
 
-1. Complete the generic probe setup (choose service, environment, version, and probe location).
 1. Select **Metric** as the probe type.
+1. Complete the generic probe setup (choose service, environment, version, and probe location).
 1. Specify a name for the metric, which will be prefixed with `dynamic.instrumentation.metric.probe.`.
 1. Select a metric type (count, gauge, or histogram).
-1. Choose the value of the metric using the Debugger expression language. For count metrics this is optional, and if you omit it, every invocation increments the count by one.
+1. Choose the value of the metric using the [Dynamic Instrumenation expression language][15]. You can use any numeric value you'd like from the execution context, such as a method parameter, local variable, a class field, or an expression that yields a numeric value. For count metrics this is optional, and if you omit it, every invocation increments the count by one.
 
 {{< img src="dynamic_instrumentation/metric_probe.png" alt="Creating a Dynamic Instrumentation metric probe" >}}
 
+### Span probes
+
+A *span probe* emits a span when a method is executed. 
+
+You can use a *span probe* as a more efficient alternative to [creating new spans with Custom Instrumentation][13]. If the method throws an exception, the details of the exception will be associated with the newly created span's `error` tag.
+
+To create a span probe:
+
+1. Select **Span** as the probe type.
+1. Complete the generic probe setup (choose service, environment, version, and probe location). 
+
+
+### Span tag probes
+
+A *span tag* probe adds a tag value to an existing span. You have two options either adding a tag to the currently active span or adding a tag to the service entry span.
+Note: internal spans are not indexed by default and thus might not be searchable in the APM trace UI.  
+
+You can use a *span tag probe* as a more efficient alternative to [using Custom Instrumentation to addings tags in code][14].
+
+To create a span tag probe:
+
+1. Select **Span Tag** as the probe type.
+1. Complete the generic probe setup (choose service, environment, version, and probe location). 
+1. Specify a name for the tag.
+1. Specify the value of the tag using the [Dynamic Instrumentation expression language][15]. 
+1. Optionally define a condition using the Dynamic Instrumentation expression language. The tag will only be added when the expression evaluates to true.
+1. Optionally add additional tags, each with their own name, expression, and optional condition.
```

Comment:
Do we want to say something if the conditions are overlapping: will both tags applied for example or do we stop after the first condition is met?

Reply:
The conditions are per-tag, and I think @shatzi made changes to the UX to make this very clear, so I don't think it necessarily needs explaining - any feedback on this @shatzi ?

Reply:
let look at usage and recording to see if this is a problem. until now I don't see it was an issue and Datadog support engineers help the user to setup on traiil.